lib/rouge/lexers/perl.rb

```diff
@@ -90,8 +91,12 @@ def self.detect?(text)
         rule %r[s\((\\\\|\\\)|[^\)])*\)\s*], re_tok, :balanced_regex
 
         rule %r(m?/(\\\\|\\/|[^/\n])*/[gcimosx]*), re_tok
-        rule %r(m(?=[/!\\{<\[\(@%\$])), re_tok, :balanced_regex
-        rule %r(((?<==~)|(?<=\())\s*/(\\\\|\\/|[^/])*/[gcimosx]*),
+        rule %r(m(?=[/!,\\{<\[\(@%\$])), re_tok, :balanced_regex
+
+        # Perl allows any non-whitespace character to delimit
+        # a regex when `m` is used.
+        rule %r(m(\S).*\1[msixpodualngc]*), re_tok
```

Comment:
`^[^$]?m(\S).*\1[msixpodualngc]*` seems to work: http://rubular.com/r/leuEDyo2s4